File: mobile/lib/models/app_branding.dart
class AppBranding {
  final String? appName;
  final String? tagline;
  final String? logoPath;
  final String? logoUrl;

  const AppBranding({
    this.appName,
    this.tagline,
    this.logoPath,
    this.logoUrl,
  });

  factory AppBranding.fromJson(Map<String, dynamic> json) {
    return AppBranding(
      appName: json['app_name'] as String?,
      tagline: json['tagline'] as String?,
      logoPath: json['logo_path'] as String?,
      logoUrl: json['logo_url'] as String?,
    );
  }
}

File: app/Models/AppBranding.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Storage;

class AppBranding extends Model
{
    protected $fillable = [
        'app_name',
        'tagline',
        'logo_path',
    ];

    public static function current(): self
    {
        $branding = static::query()->first();

        if ($branding) {
            return $branding;
        }

        return static::query()->create([
            'app_name' => 'LocalApp',
            'tagline' => 'LOYALTY PROGRAM',
            'logo_path' => null,
        ]);
    }

    public function logoUrl(): ?string
    {
        if (blank($this->logo_path)) {
            return null;
        }

        $path = ltrim((string) $this->logo_path, '/');

        // Prefer the app media route used by the mobile client.
        return secure_url('media/'.$path);
    }

    public function toApiArray(): array
    {
        return [
            'app_name' => $this->app_name,
            'tagline' => $this->tagline,
            'logo_path' => $this->logo_path,
            'logo_url' => $this->logoUrl(),
        ];
    }

    public function deleteLogoFile(): void
    {
        if (blank($this->logo_path)) {
            return;
        }

        $path = ltrim((string) $this->logo_path, '/');
        if (Storage::disk('public')->exists($path)) {
            Storage::disk('public')->delete($path);
        }
    }
}
